src/identify/index.js

```diff
@@ -90,10 +96,12 @@ class IdentifyService {
     const pushes = connections.map(async connection => {
       try {
         const { stream } = await connection.newStream(MULTICODEC_IDENTIFY_PUSH)
+        const signedPeerRecord = await this._getSelfPeerRecord()
```

Comment:
Get this once instead of in the map function. Also, we should do the same thing with the listenAddrs and protocols to avoid doing that for X connections.